File: JS/Easy/Trees/invertTree.js
/** 226
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function (root) {
  if (!root) return root;
  let traverse = (curr) => {
    //swap
    [curr.left, curr.right] = [curr.right, curr.left];
    curr.left && traverse(curr.left);
    curr.right && traverse(curr.right);
  };
  traverse(root);
  return root;
};

File: C#/DSA/Trees/invertTree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class SolutionI
{
    public TreeNode InvertTree(TreeNode root)
    {
        if (root is null) return root;
        //let's do in order traversal and swap left with right
        void Traverse(TreeNode curr)
        {
            //swap
            (curr.left, curr.right) = (curr.right, curr.left);
            if (curr.left is not null) Traverse(curr.left);
            if (curr.right is not null) Traverse(curr.right);
        }

        Traverse(root);
        return root;
    }
}